docs/ingest/README.md

```diff
@@ -0,0 +1,182 @@
+together potentially many different sources and attempts to produce a
+single tile with multiple bands. The ingest process consumes a block of
+JSON which specifies the details of exactly how a given ingest should
+occur. Further detail can be found below.  
+
+## Table of Contents
+
+* [The Ingest Process](#the-ingest-process)
+  * [Building the jar](#building-the-jar)
+  * [Running an Ingest](#running-an-ingest)
+* [Example JSON](#example-json)
+
+## The Ingest Process
+
+Each ingest is kicked off with `spark-submit` rather than `java -jar` so
+that all jars necessary for distributed processing are provided at
+runtime. This means, however, that a working Spark installation must be
+available wherever the process is to be run.  
+
+### Building the Jar
+
+Through docker, the environment needed to build and run the
+ingest are provided. To build the ingest jar, simply run
+`scripts/cibuild` or directly call
+`docker-compose run --no-deps --rm --entrypoint ./sbt app-server ingest/assembly`.  
+
+### Running an Ingest
+
+Once the ingest jar has been built, the ingest can be run locally in a
+couple of different ways. In the demonstration that follows, we'll use
+`localJob.json` which attempts a filesystem based ingest - both source
+and destination - but which requires no special credentials and runs
+fairly quickly due to the limited size of the source tiles.  
+
+A one-off local ingest which attempts to carry out the entire ingest job
+inside the driver (rather than serializing portions of the program for
+execution elsewhere) by calling:
+
+```bash
+docker-compose -f docker-compose.spark.yml run spark-driver \
+               --driver-memory 2G \
+               /opt/rf/jars/rf-ingest.jar -t -j file:///opt/rf/test-resources/localJob.json
+```
+
+To carry out a more complex (and production-like) ingest, multiple
+docker containers can be used to simulate a real, distributed cluster.
+First, we'll need to ensure that the cluster is running and is ready to
+take jobs. Note the docker-compose file selection. We must do this to
+ensure that Spark is avalaible.  
+
+```bash
+docker-compose -f docker-compose.spark.yml up spark-master spark-worker
+```
+
+At this point, a Spark driver (spark-master) and a Spark executor
+(spark-worker) should be running in their respective containers. The
+next step is to get the Spark master URL from this cluster so that we can
+submit jobs to it. Head over to [port 8888 on
+localhost](http://localhost:8888). There, you should be able to find the
+master spark address needed to proceed.  
+
+![Spark Master URL](./spark-master-url.png "Spark Master URL")  
+
+With the spark-master URL in hand and with the Spark cluster running,
+carrying out a real (serialization and all) Spark-based is as simple as
+running the following command:
+
+```bash
+docker-compose -f docker-compose.spark.yml run spark-driver \
+               --master <SPARK-MASTER-URL> \
```

Comment:
Turns out we were both wrong about the URL. We can use `spark://spark.services.rasterfoundry.internal:7077` all the time.